### PR TITLE
Add example of how to test WebSockets

### PR DESCRIPTION
**Pull request by davidpdrsn**
We occasionally get questions about how to test websockets so figured it was worth writing an example for.

### CODE REVIEW

File: examples/testing-websockets/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        // `Stream`
+        let (socket_write, mut test_rx) = futures::channel::mpsc::channel(1024);
+        let (mut test_tx, socket_read) = futures::channel::mpsc::channel(1024);
+
+        tokio::spawn(unit_testable_handle_socket(socket_write, socket_read));
+
+        test_tx
+            .send(Ok(Message::Text("foo".to_owned())))
+            .await
+            .unwrap();
```
Comment (programatik29) [inline examples/testing-websockets/src/main.rs:140]: Can be improved slightly like:

```rust
let (socket_write, mut test_rx) = futures::channel::mpsc::channel(1024);

let msgs = vec![Message::Text("foo".to_owned())];
let msg_stream = stream::iter(msgs.into_iter().map(Ok));

tokio::spawn(unit_testable_handle_socket(socket_write, msg_stream));
```

Comment (davidpdrsn) [reply examples/testing-websockets/src/main.rs:140]: I kinda wanna keep it simple as I think thats easier for people to adapt to their needs 🤔

Comment (programatik29) [reply examples/testing-websockets/src/main.rs:140]: Both patterns look useful. This one is more convenient for multiple messages sent at once.

Comment (programatik29) [reply examples/testing-websockets/src/main.rs:140]: This pattern can be added separately or we can keep it simple and omit it. Either looks fine.

**Pull request merged**